### ISSUE TITLE
VRAM texture compression always uses DXT5 codec

### ISSUE THREAD

**Issue by kubajz22** [godotengine/godot#58012]
### Godot version

v4.0.alpha2.official [79077e6c1]

### System information

Linux Mint 20.2, nvidia GTX 1650 with 470.103.01 driver

### Issue description

If you import texture in Godot 3.x in VRAM compression mode (tested with 3.4.2), the compression method chosen depends on whether the image contains transparency or not. DXT1 for RGB images and DXT5 for those with alpha channel. This doesn't seem to work in v4.0_alpha2 and the engine always chooses DXT5 and also says the texture is RGBA8 when you double click it, regardless if it actually is or not.

### Steps to reproduce

Import 2 images, one with alpha channel, one without.
Apply the compression as described above to both of them.
Godot 3 will choose different codecs, whereas Godot 4 will only use DXT5.

**Comment by Calinou** [comment]
cc @fire

**Comment by kubajz22** [comment]
> cc @fire
> 
> By the way, does DXT5 look better for opaque images compared to DXT1?

I don't think it does.. Perhaps the 'extra bits' could be used to improve the quality to some extent, but I didn't find much info on this.
What definitely is true though, is that the DXT5 takes up more VRAM.

**Comment by fire** [comment]
Sorry I misread this issue.

Currently the etcpak always does DXT5.

So we need to devise a test case that triggers an optimization opportunity for compressed RGB vs RGBA. I chose always doing RGBA (dxt5) so need some help with a test project.

**Comment by tinmanjuggernaut** [comment]
@Calinou 
> > By the way, does DXT5 look better for opaque images compared to DXT1?

@kubajz22 
> I don't think it does.. Perhaps the 'extra bits' could be used to improve the quality to some extent, but I didn't find much info on this. What definitely is true though, is that the DXT5 takes up more VRAM.

Kubajz is correct. Converting a quality 2k texture to DDS (via photoshop + intel DDS) to DXT1 and DXT5, then viewing a difference blend shows they are identical. However on the file system the first is 2.7mb and the latter 5.4mb. The memory requirements are about the same. 

This is a big deal for games with hundreds of textures like mine. DXT5 doubles the memory requirements for no gain and no option to turn it off. (For me individually, we manually convert all our textures to DDS, and it's a very good thing because the current auto conversion would screw us. We're already pushing up against 2GB of VRAM for some scenes. GD4 auto conversion would double that.)

This is still an issue in GD4 alpha 13. Can this be added to the GD4 roadmap?

@fire 
> I chose always doing RGBA (dxt5) so need some help with a test project.

There's no test project necessary. Just import two PNGs into the editor. One with alpha, one without. Import w/ vram compressed and you know it's working when the image with alpha displays as DXT5 in the inspector and the opaque one displays DXT1. This is what occurs in GD3, which was able to detect the image type.

**Comment by fire** [comment]
I support this, and I think the work is not so difficult. It's open to a pr in my opinion.

**Comment by JohanAR** [comment]
I'm having a problem with 16 bit textures looking like dithered 8 bit when setting Compress To VRAM Compressed on import, is this what could be causing it?

![Screenshot_20221110_203535](https://user-images.githubusercontent.com/4918549/201191464-fb586292-dc83-40d4-b084-d2ed0ff68f49.png)

Here's what it looks like if I use basis universal or disabled:

![Screenshot_20221110_203048](https://user-images.githubusercontent.com/4918549/201192110-e7f4c3ee-d3ef-4f99-9c0b-ba58efc4df77.png)

**Comment by tinmanjuggernaut** [comment]
When importing a png with compress to vram, it gets converted to DXT 5, which is 8bpp, not 16. You can try enabling etc2, or the other formats in the project settings/rendering/textures/VRAM compression, and try the various import options. 

Or use the settings you found.

Or preconvert the file and give Godot a format it won't convert like DDS, or a format that supports 16-bit like exr.
